go/iot/arduino/protocol.ino

```diff
@@ -0,0 +1,132 @@
+// Copyright 2018 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+//
+// This program shows how to send requests of registration and sending data to the raspberry Pi based on the protocols.
+
+#include <ArduinoJson.h>
+
+#define OK 200
+#define FAIL 404
+void setup() {
+  // Initialize Serial port
+  Serial.begin(9600);
+  while (!Serial) continue;
+}
+
+/*
+ * It would first send request of registration to the raspberry Pi.
+ * Once receive OK, it would keep sending collected data to the Pi.
+ */
+void loop() {
+   request(sendData);
+}
+
+/*
+ * The template for sending requests to the raspberry Pi.
+ * It would keep sending the commands until receive OK.
+ */
+void request(void (*func)()) {
+  int code;
+  do {
+    JsonObject* response;
+    do {
+      // TODO: Currently we hard-code the argument in the code. Would handle the problem that how to solve
+      // multiple arguments.
+      func();
+      char buffer[256];
+      readLine(buffer, 256);
+      response = parseResponse(buffer);
+    }
+    while (response == NULL);
+
+    code = (*response)["Code"];
+    const char *info = (*response)["Info"];
+
+    //Serial.print("Receive Response: Code ");
+    //Serial.print(code);
+    //Serial.print(" info: ");
+    //Serial.println(info);
+  }
```

Comment:
Consider hard-code a waiting time for now. In the future we can switch to exponential backoff.